src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -0,0 +1,540 @@
+                        backupManagementType: ServiceClientModel.BackupManagementType.AzureStorage,
+                        sourceResourceId: unregisteredStorageAccount.Properties.ContainerId,
+                        resourceGroup: vaultResourceGroupName);
+                    protectionContainerResource.Properties = prop;
+                    AzureWorkloadProviderHelper.RegisterContainer(unregisteredStorageAccount.Name,
+                        protectionContainerResource,
+                        vaultName,
+                        vaultResourceGroupName);
+                }
+            }
+
+            //inquiry
+            TriggerInquiry(vaultName, vaultResourceGroupName, storageContainerName);
+
+            //get protectable item
+            WorkloadProtectableItemResource protectableObjectResource = null;
+            protectableObjectResource = TriggerDiscovery(vaultName, vaultResourceGroupName, azureFileShareName, storageAccountName);
+
+            if (protectableObjectResource == null)
+            {
+                // Container is not discovered. Throw exception
+                string errorMessage = string.Format(
+                    Resources.DiscoveryFailure,
+                    azureFileShareName,
+                    vaultResourceGroupName);
+                Logger.Instance.WriteDebug(errorMessage);
+                Logger.Instance.WriteError(
+                    new ErrorRecord(new Exception(Resources.FileShareNotDiscovered),
+                        string.Empty, ErrorCategory.InvalidArgument, null));
+            }
+
+            return protectableObjectResource;
+        }
+
+        private WorkloadProtectableItemResource TriggerDiscovery(string vaultName, string vaultResourceGroupName,
+            string azureFileShareName, string storageAccountName)
+        {
+            WorkloadProtectableItemResource protectableObjectResource = null;
+            ODataQuery<BMSPOQueryObject> queryParam = new ODataQuery<BMSPOQueryObject>(
+                q => q.BackupManagementType
+                     == ServiceClientModel.BackupManagementType.AzureStorage);
+
+            var protectableItemList = ServiceClientAdapter.ListProtectableItem(
+                queryParam,
+                vaultName: vaultName,
+                resourceGroupName: vaultResourceGroupName);
+
+            if (protectableItemList.Count == 0)
+            {
+                //Container is not discovered
+                Logger.Instance.WriteDebug(Resources.ContainerNotDiscovered);
+            }
+
+            foreach (var protectableItem in protectableItemList)
+            {
+                AzureFileShareProtectableItem azureFileShareProtectableItem =
+                    (AzureFileShareProtectableItem)protectableItem.Properties;
+                if (azureFileShareProtectableItem != null &&
+                    string.Compare(azureFileShareProtectableItem.FriendlyName, azureFileShareName, true) == 0 &&
```

Comment:
Check with PM about how to deal with case

Reply:
tracking